src/include/sof/schedule/task.h

```diff
@@ -53,6 +55,15 @@ struct task {
 	void *private;
 };
 
+/** \brief Task type registered by pipelines. */
+struct pipeline_task {
```

Comment:
Why do we need to have a new task type. What is regular struct task missing ?

Reply:
It holds additional information, which are not needed to be included in generic task structure. Think of it like class inheritance.

Reply:
ok, I assume we are not changing any of our schedule_ APIs to use this struct ? i.e. they will still use task. This structure will only be used by pipeline ?

Btw, "registrable" seems ambiguous. Does it mean we are scheduling on IRQ ?

Reply:
API is not changed.